src/app/course/guard/instructor.guard.ts

```diff
@@ -0,0 +1,40 @@
+import {Injectable} from '@angular/core'
+import {
+    ActivatedRouteSnapshot,
+    CanActivate, Router,
+} from '@angular/router'
+import {AuthenticationService} from "@app/_services/api/authentication"
+import {Course, User} from "@app/_models"
+import {CourseService} from "@app/course/_services/course.service"
+import {map} from "rxjs/operators"
+
+@Injectable({
+    providedIn: 'root'
+})
+export class InstructorGuard implements CanActivate {
+    constructor(
+        private authenticationService: AuthenticationService,
+        private router: Router,
+        private courseService: CourseService,
+    ) {
+    }
+
+    canActivate(next: ActivatedRouteSnapshot) {
+        console.log(next.parent.params.courseId)
+        const currentUser: User = this.authenticationService.currentUserValue
+        const courseId: number = next.parent.params.courseId
+        let currentCourse: Course = null
+        return this.courseService.getCourse(courseId).pipe(
+            map(course => {
+                currentCourse = course
+                const isInstructor: boolean = currentCourse.instructor === currentUser.id
+                if (isInstructor) {
+                    return true
+                } else {
```

Comment:
This is a wrong wayof checking the instructor. You should check course registration to have type 'TA' or 'INSTRUCTOR;